src/main/kotlin/org/rust/ide/refactoring/move/common/RsMoveUtil.kt

```diff
@@ -0,0 +1,67 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.refactoring.move.common
+
+import org.rust.ide.inspections.import.insertUseItem
+import org.rust.lang.core.psi.RsBlock
+import org.rust.lang.core.psi.RsPath
+import org.rust.lang.core.psi.RsPsiFactory
+import org.rust.lang.core.psi.RsUseItem
+import org.rust.lang.core.psi.ext.*
+
+class RsMoveReferenceInfo(
+    val pathOld: RsPath,
+    val pathNew: RsPath?,
+    // == `pathOld.reference.resolve()`
+    val target: RsQualifiedNamedElement
+) {
+    val isInsideUseDirective: Boolean get() = pathOld.ancestorStrict<RsUseItem>() != null
+}
+
+fun RsPath.isAbsolute(): Boolean {
+    if (basePath().hasColonColon) return true
+    if (startsWithSuper()) return false
+
+    val basePathTarget = basePath().reference?.resolve() as? RsMod ?: return false
+    return basePathTarget.isCrateRoot
+}
+
+fun RsPath.startsWithSuper(): Boolean = basePath().referenceName == "super"
+
+// like `text`, but without spaces and comments
+val RsPath.textNormalized: String
+    get() {
+        val parts = listOfNotNull(path?.textNormalized, coloncolon?.text, referenceName, typeArgumentList?.text)
+        return parts.joinToString("")
+    }
```

Comment:
Hmm, previously I assumed that this method is used for `use`-item paths only. Is it the right assumption? If yes then `typeArgumentList` is impossible there. If not, then splitting by "::" is wrong becase a path can contain sub-paths in type arguments: `foo::bar::<baz::quux>`, as well as type qualifier `<foo::Bar>::baz`

Reply:
Currently this method is used for paths which resolve to moved file (these paths may be not in `use`-item). As far as I know it is not possible to have path resolved to `RsMod` which contains type qualifiers or type arguments. In future (in move items refactoring) we will have paths resolved to `RsItemElement`, and these paths may contain type arguments, but they will be handled separately. So lets remove `typeArgumentList` from `textNormalized` for now